cmd/installer/cmd/image.go

```diff
@@ -83,6 +84,16 @@ func runImageCmd() (err error) {
 		},
 	}
 
+	if options.ConfigSource == "" {
+		switch p.Name() {
+		case "aws", "azure", "digital-ocean", "gcp":
```

Comment:
We eventually need a better way to get platform names, but this is nothing new compared to what we are doing in other places in the code.

Reply:
This allows the user to misconfigure how userdata is retrieved. Each platform has a specific way and only a specific way to fetch userdata

Aws: https://github.com/talos-systems/talos/blob/master/internal/pkg/runtime/platform/aws/aws.go#L126
Gcp: https://github.com/talos-systems/talos/blob/master/internal/pkg/runtime/platform/gcp/gcp.go#L38
Vmware: https://github.com/talos-systems/talos/blob/master/internal/pkg/runtime/platform/vmware/vmware.go#L36

Metal is the only one that offers the flexibility to download userdata from an arbitrary location, but we don't create images for metal.

Reply:
Yeah that is a known bug for a while now. I can follow up with a PR to fix that.